File: rBike.Model/Bike.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml.Linq;

namespace rBike.Model
{
    public class Bike
    {
        public int BikeId { get; set; }

        public string Name { get; set; } = null!;

        public string BikeCode { get; set; } = null!;

        public decimal Price { get; set; }

        public int CategoryId { get; set; }

        public byte[]? Image { get; set; }

        public bool? Status { get; set; }

        public string? StateMachine { get; set; }
    }
}

File: UI/rbike_mobile/lib/models/bike.dart
import 'package:json_annotation/json_annotation.dart';

part 'bike.g.dart';

@JsonSerializable()
class Bike {
  int? bikeId;
  String? name;
  String? image;
  String? bikeCode;
  double? price;
  int? categoryId;
  String? stateMachine;

  Bike({this.bikeId, this.name, this.stateMachine});

  /// Connect the generated [_$PersonFromJson] function to the `fromJson`
  /// factory.
  factory Bike.fromJson(Map<String, dynamic> json) => _$BikeFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$BikeToJson(this);
}
